src/ERC20Reserve.sol

```diff
@@ -29,23 +37,35 @@ contract ERC20Reserve is IERC20Reserve, Ownable {
         _;
     }
 
-    function withdraw(IERC20 token, uint256 amt) public override onlyUser {
+    function withdraw(
+        IERC20 token,
+        address to,
+        uint256 amt
+    ) public override onlyUser {
         uint256 balance = withdrawable[token];
         require(balance >= amt, "Reserve: withdrawal over balance");
         withdrawable[token] = balance - amt;
-        require(token.transfer(msg.sender, amt), "Reserve: transfer failed");
-        emit Withdrawn(msg.sender, token, amt);
+        require(token.transfer(to, amt), "Reserve: transfer failed");
+        emit Withdrawn(msg.sender, token, to, amt);
     }
 
-    function deposit(IERC20 token, uint256 amt) public override onlyUser {
+    function deposit(
+        IERC20 token,
+        address from,
+        uint256 amt
+    ) public override onlyUser {
         withdrawable[token] += amt;
-        require(token.transferFrom(msg.sender, address(this), amt), "Reserve: transfer failed");
-        emit Deposited(msg.sender, token, amt);
+        require(token.transferFrom(from, address(this), amt), "Reserve: transfer failed");
```

Comment:
Why are we changing this?

For me it is important the user only will approve the DripsHub and not the Reserve.

Otherwise, the governance could change the reserve user and take all approved funds of a user from his/her EOA.

Reply:
The actual user (the EOA signer) needs to approve the `AddressId` contract anyway, not the `DripsHub`. This PR changes the address to approve by the identity contract, which only has an impact on the developers of such contracts.

This PR doesn't change the trust assumptions toward the governance. You're right about the possible attack vector in case the governance goes rogue, but in current setup it can just upgrade DripsHub to abuse all the approvals.